src/segment.ml

```diff
@@ -180,7 +180,8 @@ type t = {
   push : bool ;
   window : int ;
   options : tcp_option list ;
-  payload : Cstruct.t ;
+  payload : Cstruct.t list ;
+  payload_len : int ;
```

Comment:
I think I understand the reasonign behind this, but we have to keep in mind that nothing ensures there's no bad actor passing a segment some payload with a wrong payload_len...